File: Domain/Entities/Auction.cs
namespace Domain.Entities
{
    public class Auction
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User CreatedBy { get; set; } = null!;
        public string Title { get; set; } = "";
        public string Description { get; set; } = "";
        public decimal Price { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string ImageUrl { get; set; }
        public bool IsOpen => EndDate > DateTime.Now;
        public bool IsActive { get; set; } = true;
        public ICollection<Bid> Bids { get; set; }
    }
}

File: Frontend/src/types/Auction.ts
export interface Auction {
  id: number;
  userId: number;
  createdBy: string;
  title: string;
  description: string;
  price: number;
  startDate: string;
  endDate: string;
  imageUrl: string;
  isOpen: boolean;
  isActive: boolean;
}

export interface CreateAuction {
    userId: number;
    title: string;
    description: string;
    price: number;
    startDate: string;
    endDate: string;
    imageUrl?: string;
}